From 5fef88eaefee82255f02dec34faa239b589944ba Mon Sep 17 00:00:00 2001
From: minglumlu <ming.lu@citrix.com>
Date: Wed, 30 Aug 2017 03:58:58 +0100
Subject: [PATCH] bond: send learning pkg when non active slave failed.

In active-active bond, a non-active slave failure doesn't trigger
the sending of learning packets. This patch fixes this issue.

Signed-off-by: minglumlu <ming.lu@citrix.com>
Signed-off-by: Ben Pfaff <blp@ovn.org>
diff --git a/ofproto/bond.c b/ofproto/bond.c
index b3e9f4dc0..9ba0e7745 100644
--- a/ofproto/bond.c
+++ b/ofproto/bond.c
@@ -1670,6 +1670,8 @@ bond_slave_lookup(struct bond *bond, const void *slave_)
 static void
 bond_enable_slave(struct bond_slave *slave, bool enable)
 {
+    struct bond *bond = slave->bond;
+
     slave->delay_expires = LLONG_MAX;
     if (enable != slave->enabled) {
         slave->bond->bond_revalidate = true;
@@ -1679,6 +1681,7 @@ bond_enable_slave(struct bond_slave *slave, bool enable)
         if (enable) {
             list_insert(&slave->bond->enabled_slaves, &slave->list_node);
         } else {
+            bond->send_learning_packets = true;
             list_remove(&slave->list_node);
         }
         ovs_mutex_unlock(&slave->bond->mutex);
